backend/notification/rate_limiter.js

```diff
@@ -0,0 +1,35 @@
+class RateLimiter {
+  constructor(time_window, maximum_requests) {
```

Comment:
nit: I see you use camelCase almost everywhere, should we change this snake_case to it or is there a reason?

Reply:
No reason, I feel inclined to use the snake case when the naming is longer. I will revert all to camelCase